internal/config/types.go

```diff
@@ -32,3 +32,9 @@ type ListenerConfig struct {
 	PublishTo      string `json:"publish-to" yaml:"publish-to"`
 	SubscribeTo    string `json:"subscribe-to" yaml:"subscribe-to"`
 }
+
+// Meshsync configuration controls the resources meshsync produces and consumes
+type MeshsyncConfig struct {
+	PipelineConfigs map[string]PipelineConfigs `json:"pipeline-configs,omitempty" yaml:"pipeline-configs,omitempty"`
+	ListenerConfigs map[string]ListenerConfig  `json:"listener-config,omitempty" yaml:"listener-config,omitempty"`
```

Comment:
Here should be more clear. Please write down more detail of the data structure, why they should embed into a map like this? :

PipelineConfigs is map for {key:PipelinsConfigs} 
ListenerConfigs is a map for {key: ListenerConfig}

Reply:
Thank you for this feedback, for this change I am mimicking this [default configs](https://github.com/meshery/meshsync/blob/master/internal/config/default_config.go) . Could you further explain what your expectations are so that I can make appropriate changes

Reply:
Here I want to make sure. Does `PipelineConfigs map[string]PipelineConfigs` make sense? It will cause misunderstanding.